src/ProjectOrigin.Electricity.Client/ElectricityCommandBuilder.cs

```diff
@@ -52,8 +52,8 @@ Key issuingBodySigner
             CertificateId = federatedId,
             Period = inteval.ToProto(),
             GridArea = gridArea,
-            GsrnCommitment = gsrn.ToProtoCommitment(),
-            QuantityCommitment = quantity.ToProtoCommitment(),
+            GsrnHash = ByteString.CopyFrom(SHA256.HashData(BitConverter.GetBytes(gsrn))),
```

Comment:
And here the GSRN nr is hashed in order to protect this piece of information instead of a shielded value. Awesome.